dashboard/app/models/concerns/solution_blocks.rb

```diff
@@ -41,6 +41,31 @@ def blocks_match?(toolbox_block, solution_block)
       stripped_solution_block == strip_toolbox_block(toolbox_block)
   end
 
+  def get_solution_blocks(create_for_toolbox=true)
+    solution = Nokogiri::XML(properties['solution_blocks'])
+
+    # flatten
+    solution_blocks = solution.xpath('//block').to_a
+
+    if create_for_toolbox
+      # strip out blocks that shouldn't be in a toolbox
+      solution_blocks.reject! { |block|
+        block.get_attribute('type') == 'when_run' ||
+            block.get_attribute('type') == 'procedures_defnoreturn'
```

Comment:
Can the filtering here be the same as the filtering in add_missing_toolbox_blocks?